src/Compilers/Core/Portable/Syntax/AbstractWarningStateMap.cs

```diff
@@ -87,5 +87,13 @@ public int CompareTo(WarningStateMapEntry other)
                 return this.Position - other.Position;
             }
         }
+
+        protected class WarningStateMapEntryComparer : System.Collections.IComparer
```

Comment:
@praeclarum the predominant singleton pattern is `static readonly` fields.  In this case:

``` C#
internal static readonly WarningStateMapEntryComparer Instance = new WarningStateMapEntryComparer();
```